src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java

```diff
@@ -102,32 +111,72 @@ public UserProfile create(UserProfileCreationData profileData, String origin) {
     public UserProfile reInviteUser(UserProfileCreationData profileData) {
 
-        Optional<UserProfile>  optionalExistingUserProfile = userProfileRepository
+        Optional<UserProfile> optionalExistingUserProfile = userProfileRepository
                 .findByEmail(profileData.getEmail().toLowerCase());
         UserProfile userProfile = validationHelperService.validateReInvitedUser(optionalExistingUserProfile);
+        String emailKey = "email:";
+        String email = profileData.getEmail();
+        String emailSearchQuery = emailKey + email;
+        Map<String, String> formParams = new HashMap<>();
+        formParams.put("query", emailSearchQuery);
+        Response response = idamFeignClient.getUserFeed(formParams);
+        logIdamResponse(response);
+        if (response.status() == 200) {
+            ResponseEntity<Object> responseEntity = JsonFeignResponseUtil.toResponseEntity(response,
+                    new TypeReference<Set<IdamFeignClient.User>>() {
+                    });
+            Set<IdamFeignClient.User> users = (Set<IdamFeignClient.User>) Objects.requireNonNull(
+                    responseEntity.getBody());
+            if (!CollectionUtils.isEmpty(users)) {
+                IdamFeignClient.User user;
+                if ((user = users.stream().findFirst().orElse(null)) != null
+                        && !userProfile.getIdamId().equals(user.getId())) {
+                    log.info("Updating Idam Id in user profile");
+                    userProfile.setIdamId(user.getId());
+                    userProfile.setIdamRegistrationResponse(HttpStatus.CREATED.value());
+                    userProfile.setStatus(IdamStatus.ACTIVE);
+                    saveUserProfile(userProfile);
+                    return userProfile;
+                }
+            }
+        }
         return registerReInvitedUserInSidam(profileData, userProfile);
     }
 
     private UserProfile registerReInvitedUserInSidam(UserProfileCreationData profileData, UserProfile userProfile) {
 
+
         final IdamRegistrationInfo idamRegistrationInfo
                 = idamService.registerUser(createIdamRegistrationRequest(profileData, userProfile.getIdamId()));
         if (idamRegistrationInfo.isSuccessFromIdam()) {
             mapUpdatableFieldsForReInvite(profileData, userProfile);
             userProfile.setIdamRegistrationResponse(idamRegistrationInfo.getIdamRegistrationResponse().value());
             saveUserProfile(userProfile);
-            persistAudit(idamRegistrationInfo.getStatusMessage(), idamRegistrationInfo.getIdamRegistrationResponse(),
-                    userProfile);
+            userProfile.setIdamRegistrationResponse(201);
+            persistAudit(idamRegistrationInfo.getStatusMessage(),
+                    idamRegistrationInfo.getIdamRegistrationResponse(), userProfile);
         } else {
             String errorMessage = idamRegistrationInfo.isDuplicateUser() ? String.format(USER_ALREADY_ACTIVE
```

Comment:
as we discussed can you correct this error message ?

Reply:
Hi @rajkatla-hmcts This has been notified to praveen as well , agreed to your words but as per @kavithagade12 suggestion we need to decide on error message which we will take it up in next sprint seems ..Thank you appreciate your finding